src/java.base/share/classes/jdk/internal/classfile/CodeBuilder.java

```diff
@@ -386,7 +386,7 @@ default CodeBuilder trying(Consumer<BlockCodeBuilder> tryHandler,
 
         // Check for empty try block
         if (tryBlock.isEmpty()) {
-            throw new IllegalStateException("The body of the try block is empty");
+            throw new IllegalArgumentException("The body of the try block is empty");
```

Comment:
This exception should be documented in the method’s **Javadoc**.

Reply:
Fixed, thanks.